#!/usr/bin/env python
# _*_ coding:utf=8 _*_

'''
  Copyright: 2016-2018 ROS小课堂 www.corvin.cn
  Author: corvin
  Description:
    人脸定位时可以动态配置的一些参数,使用dynamic_reconfigure来实现.
  History:
    20181207: initial this file.
'''

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "face_location_tracker"
gen = ParameterGenerator()

gen.add("def_servo_x_min", int_t, 0, "default servo x min deg", 0,   0, 180)
gen.add("def_servo_x_max", int_t, 0, "default servo x max deg", 180, 0, 180)
gen.add("def_servo_y_min", int_t, 0, "default servo y min deg", 0,   0, 180)
gen.add("def_servo_y_max", int_t, 0, "default servo y max deg", 90,  0, 180)

gen.add("def_servo_move_step",  int_t, 0, "servo default move step(degree)",  1, 1, 5)
gen.add("face_tracker_move_delay", int_t, 0, "servo default move delay(ms)", 0, 0, 60)

gen.add("def_image_center_x", int_t, 0, "location face center x", 320, 160, 480)
gen.add("def_image_center_y", int_t, 0, "location face center y", 240, 120, 360)
gen.add("def_check_tolerance", int_t, 0, "location face tolerance", 65, 40, 100)
gen.add("face_detect_timeout", double_t, 0, "detect timeout will move org deg", 3.0, 0.5, 10.0)

gen.add("topic_move_servo", bool_t, 0, "If move servo by topic or service", True)

gen.add("face_tracker_speed", int_t, 0, "face tracker speed(bigger->faster)", 5, 1, 10)

exit(gen.generate(PACKAGE, "faceLocationDynamicNode", "faceLocation"))
